ambry-store/src/main/java/com/github/ambry/store/PersistentIndex.java

```diff
@@ -220,12 +220,8 @@ public int compare(IndexEntry e1, IndexEntry e2) {
       //from being compacted.
       removeIndexEntryNotBelongsToLastLogSegmentFromJournal();
       setEndOffsets();
-      if (config.storeAutoCloseLastLogSegmentEnabled) {
-        //if log segment is generated by auto close, at some point there's no index segment.
-        if (log.getEndOffset().getName().compareTo(getCurrentEndOffset().getName()) < 0) {
-          log.setActiveSegment(getCurrentEndOffset().getName());
-        }
-      } else {
```

Comment:
this part really confuses me. So after you made this change, say if we have a auto close log segment without any new entry, assuming the last log segment that has entry is 10_0 and the new auto closed log segment is 11_0. so the log.getEndOffset().getName() would be 11_0 and getCurrentEndOffset().getName() is 10_0. The log end offset will already greater than the index end offset. Then how is this possbile?

Reply:
Yeah, I'm just trying to make this logic backward compatible.  The original design without my auto close log segment changes has this kind of logic: "log.setActiveSegment(getCurrentEndOffset().getName());" But for my case, I think it should never happened. And since during the log initialization, the active segment will be set as the last log segment, so maybe we can remove this part of logic? Let me know which way you prefer.

Reply:
As I double checked, previously when index is missing, the corresponding logs should be gone. But since we have the recovery logic and logs should not be gone, instead, we should recover the index based on the log, is it correct? So I guess I can remove that part of logic and let the log initialization to set the active log segment.

Reply:
After a second thought, I realize that if the recovery failed, we need to set the active log segment to the last index segment related log segment. And if the recovery succeed and last log segment is empty, we should set the last log segment as active segment. Fixed the logic.